lib/puppetx/simp/compliance_mapper.rb

```diff
@@ -374,135 +374,181 @@ def import(filename, data)
                       end
                     end
                   end
-                  if continue
-                    if specification.key?("controls")
-
-                      specification["controls"].each do |control, subsection|
-                        if info.key?("controls")
-                          if info["controls"].include?(control)
-                            if info["controls"][control] == true
-                              contain = true
-                            else
-                              contain = false
-                            end
-                          end
-                        end
+                end
+              end
 
-                      end
-                    end
-                    if specification.key?("ces")
-
-                      specification["ces"].each do |ce|
-                        if (info.key?("ces"))
-                          if (info["ces"].key?(ce))
-                            if (info["ces"][ce] == true)
-                              contain = true
-                            else
-                              contain = false
-                              continue = false
-                            end
-                          end
-                        end
+              # Skip if we didn't find any controls to match against
+              unless found_control_match
+                @callback.debug("SKIP: '#{check}' had no matching controls")
+                next
+              end
+            end
+          end
 
-                        if @configuration_element_list.key?(ce)
-                          if @configuration_element_list[ce].key?("controls")
-                            controls = @configuration_element_list[ce]["controls"]
-                            controls.each do |control, subsection|
-
-                              if continue == true
-                                if info.key?("controls")
-                                  if info["controls"].include?(control)
-                                    contain = true
-                                  end
-                                end
-                              end
-                            end
-                          end
-                        end
+          # Now that all potential matches have been collected, we need to
+          # apply the confinement
+          specifications.delete_if do |specification|
+            delete_item = false
+
+            catch(:confine_end) do
+              if specification.key?('confine')
+                confine = specification['confine']
+
+                if confine
+                  unless confine.is_a?(Hash)
+
+                    unless specification['settings'].key?('value')
+                      if specification['telemetry'] && specification['telemetry'].first
+                        location = specification['telemetry'].first['filename']
                       end
+
+                      location ||= 'unknown'
```

Comment:
Ruby scope can be funny.  I would set `location= 'unknown'` prior to the if block.